### DISCUSSION TITLE
An option to disable the "dimension door" spell

### DISCUSSION THREAD

**Discussion by zuultek** [ihhub/fheroes2#5715] [Ideas]
Would be nice to have an option to prevent dimension door from appearing in magic guilds and pyramids, as it can ruin gameplay on some maps. Even AI seems to be having no issues with bypassing locked colored doors with it.

**Comment by jeanph01** [comment]
I think also Town Portal and Town Gate should fit in the same category... Those spells tend to make the game too easy to win.

**Comment by oleg-derevenetz** [comment]
Hi guys, you definitely should try to play "hardcore", without all this abuse of saving / loading after each wrong decision, lost battle or massive losses.

**Reply by Branikolog** [reply]
Hi all! 
I agree with @oleg-derevenetz , playing on higher difficulties means, that you're going to be playing with such drastic resource shortage, that you won't even think about building 5th level guilds. ;)

**Comment by ihhub** [comment]
Also "easy" difficulty exists just for such cases.

**Comment by zuultek** [comment]
I ended up changing the code myself to set dd's level to 6, seems to be working for now.

The post wasn't about difficulty, but about bypassing locked doors, what's with these assumptions about how i play.

**Comment by ihhub** [comment]
The original game has exactly the same behavior. The only difference is that that AI is incapable to use the spell while fheroes2 AI uses it. Adding another option is not a proper solution but rather that a quick workaround. Even in the original game some maps are easy to beat with some spells.

HoMM2 map format doesn't allow to disable certain spells. There are some maps where having certain spells gives advantages to players or AI.

**Reply by Branikolog** [reply]
The only controversial thing regarding "Dimension Door", in my opinion, spell is that it allows to jump over creatures without initiating a battle and also to jump on any object, like mountain, which allows player to steal almost every item being guarded by neutral monsters.
![image](https://user-images.githubusercontent.com/55348946/184322250-a810323d-383a-486c-8039-8d586c371608.png)

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> The only controversial thing regarding "Dimension Door", in my opinion, spell is that it allows to jump over creatures without initiating a battle and also to jump on any object, like mountain, which allows player to steal almost every item being guarded by neutral monsters. ![image](https://user-images.githubusercontent.com/55348946/184322250-a810323d-383a-486c-8039-8d586c371608.png)

Yes, that's the spirit of Heroes 2.
The player has spent time and resources to build one or more mage guilds, or to explore one or more pyramids.
Now he reaps the benefits.
This is a good strategy to develop in the game. :)
